examples/asset/alter_mesh.rs

```diff
@@ -0,0 +1,200 @@
+            Shape::Cube => Shape::Sphere,
+            Shape::Sphere => Shape::Cube,
+        }
+    }
+}
+
+#[derive(Component, Debug)]
+struct Left;
+
+fn setup(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let left_shape = Shape::Cube;
+    let right_shape = Shape::Cube;
+
+    let left_shape_model = asset_server.load(
+        GltfAssetLabel::Primitive {
+            mesh: 0,
+            primitive: 0,
+        }
+        .from_asset(left_shape.get_model_path()),
+    );
+    let right_shape_model = asset_server.load(
+        GltfAssetLabel::Primitive {
+            mesh: 0,
+            primitive: 0,
+        }
+        .from_asset(right_shape.get_model_path()),
+    );
+
+    // Add a material asset directly to the materials storage
+    let material_handle = materials.add(StandardMaterial {
+        base_color: Color::srgb(0.6, 0.8, 0.6),
+        ..default()
+    });
+
+    commands.spawn((
+        Left,
+        Name::new("Left Shape"),
+        PbrBundle {
+            mesh: left_shape_model,
+            material: material_handle.clone(),
+            transform: Transform::from_xyz(-3.0, 0.0, 0.0),
+            ..default()
+        },
+        left_shape,
+    ));
+
+    commands.spawn((
+        Name::new("Right Shape"),
+        PbrBundle {
+            mesh: right_shape_model,
+            material: material_handle,
+            transform: Transform::from_xyz(3.0, 0.0, 0.0),
+            ..default()
+        },
+        right_shape,
+    ));
+
+    commands.spawn((
+        Name::new("Point Light"),
+        PointLightBundle {
+            transform: Transform::from_xyz(4.0, 5.0, 4.0),
+            ..default()
+        },
+    ));
+
+    commands.spawn((
+        Name::new("Camera"),
+        Camera3dBundle {
+            transform: Transform::from_xyz(0.0, 3.0, 20.0).looking_at(Vec3::ZERO, Vec3::Y),
+            ..default()
+        },
+    ));
+}
+
+fn instructions(mut commands: Commands) {
+    commands
+        .spawn((
+            Name::new("Instructions"),
+            NodeBundle {
+                style: Style {
+                    align_items: AlignItems::Start,
+                    flex_direction: FlexDirection::Column,
+                    justify_content: JustifyContent::Start,
+                    width: Val::Percent(100.),
+                    ..default()
+                },
+                ..default()
+            },
+        ))
+        .with_children(|parent| {
+            parent.spawn(TextBundle::from_section(
+                "Space: swap meshes by mutating a Handle<Mesh>",
+                TextStyle::default(),
+            ));
+            parent.spawn(TextBundle::from_section(
+                "Return: mutate the mesh itself, changing all copies of it",
+                TextStyle::default(),
+            ));
+        });
+}
+
+#[derive(Default)]
+struct IsMeshScaled(bool);
+
+fn keyboard_controls(
```

Comment:
Could you split this into two systems so that they can have meaningful names?
Doing so also allows you to completely skip the `keyboard_input` param by using a `input_just_pressed` run condition :)

Reply:
Wait, that's a thing? Nice!